src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1541,6 +1543,15 @@ public static bool HasNames(this TupleExpressionSyntax tuple)
                 {
                     return true;
                 }
+
+                if (previous.IsKind(SyntaxKind.AwaitKeyword))
+                {
+                    var beforeAwait = previous.GetPreviousToken(includeSkipped: true);
```

Comment:
I would assume this would check that the 'await' parsed as a part of a statement vs. just checking previous token. You sure don't want "using await" at the start of a file (i.e. namespace using) triggering this.

Reply:
(and yes, it looks like the previous check also has a bug. :smile:)

Reply:
From another PR, I understand that recommending keywords a little loosely isn't bad. 
https://github.com/dotnet/roslyn/pull/24089
Are you saying they both should be tightened?

Reply:
This seems like this shouldnt' be necessary.  Instead of:

```c#
                 var previous = token.GetPreviousToken(includeSkipped: true);
                 if (previous.IsKind(SyntaxKind.ForKeyword) ||
                     previous.IsKind(SyntaxKind.ForEachKeyword) ||
                     previous.IsKind(SyntaxKind.UsingKeyword))
                  {
                      return true;
                  }
```

We should jsut check the parent node kind.

Reply:
i.e. if we have an open paren, see if the parent of it is a BaseForeachStatementSyntax, ForStatementSyntax, or a UsingStatementSyntax.